File: distros/J/JavonetPerlSdk/lib/Javonet/Binaries/Python/javonet/utils/connectionData/IConnectionData.py
from abc import ABC, abstractmethod
from javonet.utils.ConnectionType import ConnectionType


class IConnectionData(ABC):
    @property
    @abstractmethod
    def connection_type(self):
        pass

    @property
    @abstractmethod
    def hostname(self):
        pass

    @abstractmethod
    def serialize_connection_data(self):
        pass

File: distros/J/JavonetPerlSdk/lib/Javonet/Binaries/Nodejs/utils/connectionData/IConnectionData.js
const ConnectionType = require('../../utils/ConnectionType');

class IConnectionData {
    get connectionType() {
        throw new Error('You have to implement the method connectionType!');
    }

    get hostname() {
        throw new Error('You have to implement the method hostname!');
    }

    serializeConnectionData() {
        throw new Error('You have to implement the method serializeConnectionData!');
    }
}

module.exports = IConnectionData;